src/core/p5.Renderer2D.js

```diff
@@ -401,21 +401,26 @@ p5.Renderer2D.prototype.updatePixels = function(x, y, w, h) {
     let imgData;
     if (pixels instanceof Uint8ClampedArray) {
       if (pixels.length === pixelsState.imageData.data.length) {
-        imgData = new ImageData(pixels, pixelsState.width, pixelsState.height);
+        imgData = new ImageData(
+          pixels,
+          pixelsState.imageData.width,
+          pixelsState.imageData.height
+        );
       } else {
-        let imgArr = Array.from(pixels);
-        imgArr.length = pixelsState.imageData.data.length;
+        let imgArr = new Uint8ClampedArray(pixelsState.imageData.data.length);
+        imgArr.set(pixels);
         imgData = new ImageData(
-          Uint8ClampedArray.from(imgArr),
-          pixelsState.width,
-          pixelsState.height
+          imgArr,
+          pixelsState.imageData.width,
+          pixelsState.imageData.height
         );
       }
     } else {
       if (Array.isArray(pixels)) {
-        pixels.length = pixelsState.imageData.data.length;
+        let imgArr = new Uint8ClampedArray(pixelsState.imageData.data.length);
+        imgArr.set(pixels);
```

Comment:
I think the original implementation for this is simpler but if you just want to keep things consistent then I guess this should be fine as well.

Reply:
Yeah I thought it'll be fine this way too, if we are doing the same above. :D
As for unit tests, I checked and I think no unit tests exist for `updatePixels()`, so I guess I will write some.